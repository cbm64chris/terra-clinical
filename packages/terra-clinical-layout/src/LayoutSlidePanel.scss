/* stylelint-disable selector-class-pattern */

:local {
  $hover-target-width: var(--layout-slide-hover-target-width, 30px);

  .layout-slide-panel {
    height: 100%;
    overflow: hidden;
    position: relative;
    width: 100%;
  }

  .content,
  .panel {
    box-sizing: border-box;
    height: 100%;
    overflow: hidden;
    transition: none;
  }

  .is-animated {
    transition: transform 0.15s ease;
  }

  .content {
    float: right;
    position: relative;
    width: 100%;
    z-index: 0;
  }

  .panel {
    background-color: var(--layout-slide-panel-background-color, #fff);
    position: absolute;
    transform: translate3d(-100%, 0, 0);
    width: var(--layout-slide-panel-desktop-panel-width, 260px);
    z-index: 5000;
  }

  // Move the slider back to its origin when its on-screen.
  .is-open > .panel {
    transform: translate3d(0, 0, 0);
  }

  .hover-toggle-enabled.is-overlay {
    > .content {
      width: calc(100% - #{$hover-target-width});// allow for hover bar
    }

    > .panel {
      transform: translateX(-100%) translateX($hover-target-width);
    }

    &.is-open > .panel {
      transform: translate3d(0, 0, 0);
    }
  }

  // Due to conflicts with the RTL plugin and the use of the variable, we need to explicitly
  // specify the rtl-based transform value.
  [dir=rtl] {
    .hover-toggle-enabled.is-overlay {
      > .panel {
        transform: translateX(100%) translateX(calc(-1 * #{$hover-target-width}));
      }
    }
  }

  .panel.is-tiny {
    width: var(--layout-slide-panel-phone-panel-width, calc(100% - 50px));
  }

  .panel.is-small {
    width: var(--layout-slide-panel-tablet-panel-width, 320px);
  }

  // Overlay
  .is-open.is-overlay > .panel {
    box-shadow: 1px 0 5px var(--layout-slide-panel-panel-shadow-color, #bcbfc0);
  }

  // Squish
  .is-open.is-squish {
    > .content {
      width: calc(100% - #{var(--layout-slide-panel-desktop-panel-width, 260px)} - 1px);
    }

    > .panel {
      border-right: 1px solid var(--layout-slide-panel-panel-border-color, #d3d4d5);
      width: calc(1px + #{var(--layout-slide-panel-desktop-panel-width, 260px)});
      transition: none;
    }
  }
}
