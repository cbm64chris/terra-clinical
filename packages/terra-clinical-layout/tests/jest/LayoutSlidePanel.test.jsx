import React from 'react';

import LayoutSlidePanel from '../../src/_LayoutSlidePanel';

  // /**
  //  * Whether or not the panel transitions should be animated.
  //  */
  // isAnimated: PropTypes.bool,
  // /**
  //  * Whether or not the panel should be displayed.
  //  */
  // isOpen: PropTypes.bool,
  // /**
  //  * Whether or not panel toggling is enabled.
  //  */
  // isToggleEnabled: PropTypes.bool,
  //   /**
  //  * The element to display in the main content area.
  //  */
  // children: PropTypes.element,
  // /**
  //  * The style of panel presentation. One of `overlay`, `squish`.
  //  */
  // panelBehavior: PropTypes.oneOf(['overlay', 'squish']),
  // /**
  //  * The component to display in the panel content area.
  //  */
  // panelContent: PropTypes.node,
  // /**
  //  * Current breakpoint size.
  //  */
  // size: PropTypes.string.isRequired,
  // /**
  //  * The function called when panel state changes are desired.
  //  */
  // onToggle: PropTypes.func,
  // /**
  //  * String to display on menu hover target.
  //  */
  // toggleText: PropTypes.string,

describe('LayoutSlidePanel', () => {
  it('should render without optional props', () => {
    const result = shallow((
      <LayoutSlidePanel
        size="medium"
      />
    ));
    expect(result).toMatchSnapshot();
  });

  it('should render with provided props', () => {
    const result = shallow((
      <LayoutSlidePanel
        isAnimated
        isOpen
        isToggleEnabled
        panelBehavior="overlay"
        panelContent={(
          <div>Panel Content</div>
        )}
        size="medium"
        onToggle={() => {}}
        toggleText="Test Text"
      >
        <div>Child</div>
      </LayoutSlidePanel>
    ));

    expect(result).toMatchSnapshot();
  });

  it('should render when small', () => {
    const result = shallow((
      <LayoutSlidePanel
        isAnimated
        isOpen
        isToggleEnabled
        panelBehavior="overlay"
        panelContent={(
          <div>Panel Content</div>
        )}
        size="small"
        onToggle={() => {}}
        toggleText="Test Text"
      >
        <div>Child</div>
      </LayoutSlidePanel>
    ));

    expect(result).toMatchSnapshot();
  });

  it('should render when tiny', () => {
    const result = shallow((
      <LayoutSlidePanel
        isAnimated
        isOpen
        isToggleEnabled
        panelBehavior="overlay"
        panelContent={(
          <div>Panel Content</div>
        )}
        size="tiny"
        onToggle={() => {}}
        toggleText="Test Text"
      >
        <div>Child</div>
      </LayoutSlidePanel>
    ));

    expect(result).toMatchSnapshot();
  });
});
