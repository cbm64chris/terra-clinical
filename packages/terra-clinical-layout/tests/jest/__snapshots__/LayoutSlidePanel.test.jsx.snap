// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutSlidePanel should render when small 1`] = `
<div
  className="layout-slide-panel is-open is-overlay"
>
  <div
    aria-hidden={null}
    className="panel is-small is-animated"
  >
    <HoverTarget
      hoverIsEnabled={false}
      isEnabled={false}
      isOpen={true}
      onClick={[Function]}
      onHoverOff={[Function]}
      onHoverOn={[Function]}
      text="Test Text"
    >
      <div>
        Panel Content
      </div>
    </HoverTarget>
  </div>
  <div
    className="content"
  >
    <Overlay
      backgroundStyle="dark"
      isOpen={true}
      isRelativeToContainer={true}
      isScrollable={false}
      onRequestClose={[Function]}
    />
    <div>
      Child
    </div>
  </div>
</div>
`;

exports[`LayoutSlidePanel should render when tiny 1`] = `
<div
  className="layout-slide-panel is-open is-overlay"
>
  <div
    aria-hidden={null}
    className="panel is-tiny is-animated"
  >
    <HoverTarget
      hoverIsEnabled={false}
      isEnabled={false}
      isOpen={true}
      onClick={[Function]}
      onHoverOff={[Function]}
      onHoverOn={[Function]}
      text="Test Text"
    >
      <div>
        Panel Content
      </div>
    </HoverTarget>
  </div>
  <div
    className="content"
  >
    <Overlay
      backgroundStyle="dark"
      isOpen={true}
      isRelativeToContainer={true}
      isScrollable={false}
      onRequestClose={[Function]}
    />
    <div>
      Child
    </div>
  </div>
</div>
`;

exports[`LayoutSlidePanel should render with provided props 1`] = `
<div
  className="layout-slide-panel is-open is-overlay hover-toggle-enabled"
>
  <div
    aria-hidden={null}
    className="panel is-animated"
  >
    <HoverTarget
      hoverIsEnabled={true}
      isEnabled={false}
      isOpen={true}
      onClick={[Function]}
      onHoverOff={[Function]}
      onHoverOn={[Function]}
      text="Test Text"
    >
      <div>
        Panel Content
      </div>
    </HoverTarget>
  </div>
  <div
    className="content"
  >
    <Overlay
      backgroundStyle="clear"
      isOpen={true}
      isRelativeToContainer={true}
      isScrollable={false}
      onRequestClose={[Function]}
    />
    <div>
      Child
    </div>
  </div>
</div>
`;

exports[`LayoutSlidePanel should render without optional props 1`] = `
<div
  className="layout-slide-panel is-overlay"
>
  <div
    className="content"
  >
    <Overlay
      backgroundStyle="clear"
      isOpen={false}
      isRelativeToContainer={true}
      isScrollable={false}
    />
  </div>
</div>
`;
